### DISCUSSION TITLE
419 Page Expired + index.js error on login /admin/login form submission

### DISCUSSION THREAD

**Discussion by giovanniledda** [filamentphp/filament#14089] [Help] [answered]
### Package

Other

### Package Version

v3.2.108

### How can we help you?

Hi!

I'm running my Filament site on a staging environment, and I encountered this error: when I submit the login form, I get a "419 page expired" response from the _/livewire/update_ request. Then, in the JS console, I see this::

![Screenshot 2024-08-29 alle 12 29 36](https://github.com/user-attachments/assets/619b5317-a2cb-479b-877d-a37a463bceee)

![Screenshot 2024-08-29 alle 12 29 43](https://github.com/user-attachments/assets/0a14bb69-d55e-4d2e-8833-a4f22d7a3526)

Notes:
- same application, on local environment, works fine.
- here some .env params:

```
SESSION_DRIVER=database
SESSION_LIFETIME=120
SESSION_ENCRYPT=false
SESSION_PATH=/
SESSION_DOMAIN=my-site.staging.my-domain.it
SESSION_SAME_SITE=null
SESSION_SECURE_COOKIE=false
```

- the site is running on a docker container, with Nginx, here my Nginx conf:

```
server {

    server_name my-site.staging.my-domain.it;
    root /var/www/laravel_phpfpm/my-site/public;

    location ~ [^/]\.php(/|$) {
        try_files $uri =404;
        fastcgi_split_path_info ^(.+?\.php)(|/.*)$;
        include fastcgi_params;
        fastcgi_param HTTP_PROXY "";
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param PATH_INFO $fastcgi_path_info;
        fastcgi_param QUERY_STRING $query_string;
        fastcgi_intercept_errors on;
        fastcgi_pass 127.0.0.1:19008;
        #fastcgi_buffer_size 32k;
        fastcgi_read_timeout 3600s;
    }

     location / {
         try_files $uri  /index.php$is_args$args;
     }

    location /.well-known {
        root /var/www/html;
    }

#     location ^~ /.well-known {
#        allow all;
#        root /var/www/laravel_phpfpm/my-site/public;
#    }

    location ^~ /filament {
        try_files $uri $uri/ /index.php?$query_string;
    }

#     location ~* \.(jpg|jpeg|gif|png|webp|svg|woff|woff2|ttf|css|js|ico|xml)$ {
#         try_files $uri /index.php?$query_string;
#         access_log        off;
#         log_not_found     off;
#         expires           14d;
#     }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/my-site.staging.my-domain.it/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/my-site.staging.my-domain.it/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
```

- Laravel Framework 11.21.0
- PHP 8.2.18

**Comment by CristinaQi** [comment]
I'm experimenting the same error, no solutions so far

**Comment by giovanniledda** [comment, marked as answer]
SOLUTION:  In one of my configuration files, I've inserted a space before the <? beginning of a PHP file. This was before any headers were set, and so they weren't allowed to be changed (meaning no cookies could be set) since data had already been written to the output stream. So if you're searching for why this is happening to you, make sure you look for that.

**Comment by hifshan** [comment]
If you not found the configuration files that beginning with space in your project, try to execute this command
php artisan filament:optimize

i spent more than 6 hours to solve this problem. trying to add verifyCsrfToken etc and it doesn't working before optimize it.

**Reply by leek** [reply]
@hifshan THANK YOU. I fought this random issue for hours and this was the solution.
